BEPs/BEP-346.md

```diff
@@ -0,0 +1,121 @@
+
+## 2. Motivation
+Currently, users are required to delete and recreate objects in order to replace their content in Greenfield. 
+This process leads to the loss of associated policies and change to the object's ID, requiring users to re-enforce policies 
+and trigger any mirroring processes again, refer to [Resource Mirror](https://docs.bnbchain.org/greenfield-docs/docs/faq/mirroring-faqs/#what-is-mirroring) for more details.  The motivation behind this proposal is to address these issues and provide a 
+more efficient and seamless object update process.
+
+## 3. Status
+This BEP is under progress
+
+## 4. Specification
+
+### 4.1 Requirements
+- Atomicity: Atomicity refers to the ability to ensure that updates to an object either all succeed or all fail, guaranteeing that the object remains in a consistent state.
+- Consistency: Ensures that an object remains in a valid and predictable state after a successful update. Object metadata on the Greenfield Blockchain and the data stored on the Storage Provider will be maintained to ensure users can retrieve correct object data at any state of the updating process.
+
+### 4.2 Implementation
+
+#### 4.2.1. Change To Current Model
+- [ObjectInfo](https://docs.bnbchain.org/greenfield-docs/docs/guide/storage-provider/modules/common/proto/#objectinfo-proto):
+  ObjectInfo is used for object metadata stored in Greenfield. Below fields are added.
+  - `updating`: Indicate whether an object is being updated. It means the user has initiated an update request, and the SP has not committed the change to Greenfield yet.
+  - `updated_at`: The timestamp that the user initiated to update the object. The storage price is determined by it, and a new store fee needs to be calculated.
+  - `version`: The current version of the object. It can be used to locate the correct version of object data in SP.
+
+#### 4.2.1. New Model And Message
+- `ShadowObjectInfo`: It is introduced to store user updated object metadata, the life cycle for a shadowObjectInfo is 
+  started when users trigger the update request on Greenfield, once SP has the updated object data stored and confirm 
+  on Greenfield, it will be removed.
+  - `operator`: The account who initiates update object request, can be the owner or any account that has been granted permission to update object content.
+  - `id`: The id of the object, it is the same as the id in ObjectInfo.
+  - `payload_size`: The updated object size.
+  - `checksums`: The updated checksums of pieces stored in storage providers.
+  - `updated_at`: The timestamp that user initiated to update object request.
+  - `version`: The expected next version of the object.
+- `MsgUpdateObjectContent`: A new message for users' update object content transaction. The new object's payload_size,
+  checksums along with approval from primary SP's approval need to be specifcied.
+```protobuf
```

Comment:
Please use the same style as `ShadowObjectInfo`.  Do not put codes here directly.